src/storage-manager.coffee

```diff
@@ -67,6 +61,9 @@ class _StorageManager
     collection.on 'remove', (model) ->
       model.invalidateCache()
 
+    collection.on 'change', (model) =>
+      @_invalidateCache(model.brainstemKey) unless model.isNew()
```

Comment:
So when is the cache being used then?

Reply:
I've opted to not bust caches between associations. I think this will be a decent approximation. There will still be instances of `cache: false` when the data fetched relies heavily on associations. However, I think we need a few case studies to determine the best course of action for inferring association chains between collections and when to bust the caches across associations.